gui/src/pages/NewProjectConfig.svelte

```diff
@@ -28,6 +31,9 @@
   let dirPromise: Promise<ReadDirResult> = api.kernel
     .getUserHomeDir()
     .then((dir) => {
+      if (root) {
+        dir = root;
+      }
       setWorkingDirectory(dir);
       homeDirectory = dir;
       return dirPromise;
```

Comment:
This is really strange... In general, the side effects in this callback seem like they are subverting the Svelte reactive mechanisms.

Reply:
Do you man the new 3 lines or the entire mechanism @brandonbloom

Reply:
The side effects in this callback in general.